src/udev/udev-builtin-input_id.c

```diff
@@ -193,7 +195,7 @@ static bool test_pointers(struct udev_device *dev,
         is_pointing_stick = test_bit(INPUT_PROP_POINTING_STICK, bitmask_props);
         stylus_or_pen = test_bit(BTN_STYLUS, bitmask_key) || test_bit(BTN_TOOL_PEN, bitmask_key);
         finger_but_no_pen = test_bit(BTN_TOOL_FINGER, bitmask_key) && !test_bit(BTN_TOOL_PEN, bitmask_key);
-        has_mouse_button = test_bit(BTN_LEFT, bitmask_key);
+        for (btn = BTN_LEFT; btn < BTN_JOYSTICK && !(has_mouse_button = test_bit(btn, bitmask_key)); btn++);
```

Comment:
Please add a comment above the loop saying what it does. It's not entirely obvious at first glance. And an empty line before the comment and after the loop.

The for-loop style is a bit unusual, but is seems to work here better then alternatives, so OK.